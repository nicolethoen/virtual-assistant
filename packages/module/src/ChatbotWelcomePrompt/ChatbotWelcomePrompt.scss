// ============================================================================
// Chatbot Layout - Chat - Welcome
// ============================================================================
.pf-chatbot--layout--welcome {
  padding-block-end: var(--pf-t--global--spacer--lg);
  padding-block-start: var(--pf-t--global--spacer--lg);

  .pf-v6-c-content--h1 {
    --pf-v6-c-content--h1--FontSize: var(--pf-t--global--font--size--3xl); // larger than any of our semantic tokens
    --pf-v6-c-content--h1--FontWeight: var(--pf-t--global--font--weight--400);
    --pf-v6-c-content--h1--MarginBlockEnd: 0;
  }

  .pf-chatbot__hello {
    color: var(--pf-t--global--color--brand--default);
  }

  .pf-chatbot__question {
    color: var(--pf-t--global--text--color--subtle);
  }
}
