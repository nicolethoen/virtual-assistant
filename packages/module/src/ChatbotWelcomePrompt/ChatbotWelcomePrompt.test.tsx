import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ChatbotWelcomePrompt from './ChatbotWelcomePrompt';
import userEvent from '@testing-library/user-event';

describe('ChatbotWelcomePrompt', () => {
  it('should render welcome prompt', () => {
    const { container } = render(
      <ChatbotWelcomePrompt title="Hello, Chatbot User" description="How may I help you today?" />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly', () => {
    render(<ChatbotWelcomePrompt title="Hello, Chatbot User" description="How may I help you today?" />);
    expect(screen.getByText('Hello, Chatbot User')).toBeTruthy();
    expect(screen.getByText('How may I help you today?')).toBeTruthy();
  });
  it('should render prompts with titles correctly', () => {
    render(
      <ChatbotWelcomePrompt
        title="Hello, Chatbot User"
        description="How may I help you today?"
        prompts={[{ title: 'Topic 1' }]}
      />
    );
    expect(screen.getByText('Topic 1')).toBeTruthy();
  });
  it('should render prompts with messages correctly', () => {
    render(
      <ChatbotWelcomePrompt
        title="Hello, Chatbot User"
        description="How may I help you today?"
        prompts={[{ title: 'Topic 1', message: 'Helpful prompt for Topic 1' }]}
      />
    );
    expect(screen.getByText('Helpful prompt for Topic 1')).toBeTruthy();
  });
  it('should render prompts with onClick correctly', async () => {
    const spy = jest.fn();
    render(
      <ChatbotWelcomePrompt
        title="Hello, Chatbot User"
        description="How may I help you today?"
        prompts={[{ title: 'Topic 1', message: 'Helpful prompt for Topic 1', onClick: spy }]}
      />
    );
    await userEvent.click(screen.getByRole('button', { name: /Topic 1/i }));
    expect(spy).toHaveBeenCalled();
  });
  it('should apply className appropriately', () => {
    render(
      <ChatbotWelcomePrompt
        title="Hello, Chatbot User"
        description="How may I help you today?"
        className="test"
        testId="welcome-prompt"
      />
    );
    const element = screen.getByTestId('welcome-prompt');
    expect(element).toHaveClass('test');
  });
});
