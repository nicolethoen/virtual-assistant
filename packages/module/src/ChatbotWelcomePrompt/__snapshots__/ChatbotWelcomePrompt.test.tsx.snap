// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatbotWelcomePrompt should render welcome prompt 1`] = `
<div>
  <div
    class="pf-chatbot--layout--welcome "
  >
    <h1
      class="pf-v6-c-content--h1"
      data-ouia-component-id="OUIA-Generated-Content-1"
      data-ouia-component-type="PF6/Content"
      data-ouia-safe="true"
      data-pf-content="true"
    >
      <span
        class="pf-chatbot__hello"
      >
        Hello, Chatbot User
      </span>
      <br />
      <span
        class="pf-chatbot__question"
      >
        How may I help you today?
      </span>
    </h1>
    <div
      class="pf-chatbot__prompt-suggestions"
    />
  </div>
</div>
`;
